File: demo_PES.py
import time
import numpy as np
import matplotlib.pyplot as plt
import matlab.engine

eng = matlab.engine.start_matlab()

data = eng.propagating_energy_sources();

dirname   = 'PES-rn'
bfilename = dirname + '/pes'
eng.mkdir( dirname )

myOpts = eng.struct( 'cmethod', 'phase',
        'spans', 21, 'detrend', 2,
        'errorRate', 0.1,
        'boolParfor', 0,
        'saveData', 0, 'boolUseSavedData', 0,
        'bfilename', bfilename);


eng.tic;
Components, Clusters, ClusterInfo, SDFInfo = eng.pasf(data, 4, myOpts, nargout=4)
eng.toc;

Components = np.array(Components._data).reshape(Components.size, order='F')
print( Components.shape )
d1, d2, d3, d4 = Components.shape

for i in range(d3):
    fig = Components[:,:,i,0]
    for c in range(1, d4):
        fig = np.hstack((fig, Components[:,:,i,c]))

    plt.imshow(fig)
    plt.show(block=False)
    time.sleep(0.5)
    plt.close()

File: demo_PES.m

data = propagating_energy_sources();

dirname   = strcat( 'PES' );
bfilename = strcat( dirname, '/pes' );
mkdir( dirname );

myOpts = struct( 'cmethod', 'phase', ...
	'spans', 21, 'detrend', 2, ...
	'errorRate', 0.1, ...
	'boolParfor', false, ... 
	'saveData', 0, 'boolUseSavedData', 0, ...
	'bfilename', bfilename);

tic;
Z = pasf(data, 4, myOpts); 
toc;

[d1, d2, d3, d4] = size(Z);
rng = prctile( Z(:), [1 99] ) + [-eps eps];
for f = 1:d3,
	frame = Z(:,:,f,1);
	for c = 2:d4,
		frame = horzcat(frame, Z(:, :, f, c) );
	end
	imagesc( frame, rng);  
	title('Component #1, Component #2, Component #3, Component #4, Remainder, and Input Signal');
	daspect([1 1 1]);
	axis off;
	pause(0.5);
end
